reference/multipart.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: page
+title: Multipart Encoding
+---
+
+Praxis has built-in support for handling "multipart/form-data" (link to RFC?) encoded requests and responses, in the form of the `Praxis::Types::MultipartArray` type, and the `Praxis::Responses::MultipartOk` response.
+
+
+## Praxis::Types::MultipartArray
+
+The `MultipartArray` type is an `Attributor::Collection` of `Praxis::MultipartPart` members that allows you to specify the handling of each part of a multipart request or response body.
+
+### Definition
+
+* `part 'name'`
+* `part /name/`
+* `part 'email', multiple: true`
+
+* defining parts that are files:
+  * simple: `part 'file', filename: true`
+  * simpler: `file 'file'`
+  * with filename string validation:
+    * `file 'file' { filename values: ['thumb.jpg'] }`
+    * `file 'file' { filename regex: /^img-/ }`
+  * with a filename of type other than String
+    * `file 'daily' { filename DateTime }`
+
+Example: 
+
+{% highlight ruby %}
+class ImageUpload < Praxis::Types::MultipartArray
+
+  # Image author, loaded (and validated) as pre-defined Author type  
+  part 'author', Author, 
+    required: true, 
+    description: 'Authorship information'  
+  
+  # Set of tags, as strings. May be specified more than once.
+  part 'tags', String
+    multiple: true, 
+    description: 'Category name. May be given multiple times'
+
+  # Any part whose name ends in '_at' should be a DateTime
+  part /_at$/, description: 'Timestamp information for the set of files' do
+    payload DateTime
+  end
+
+  # The parser will save uploaded file as a Ruby Tempfile.
+  # Note: This maps to the Attributor::Tempfile type.
+  part 'image', Tempfile, 
+    required: true
+    filename: true, 
+    description: 'Image to upload'
+
+  # Ensure the incoming thumbnail is a jpeg with proper filename'
+  file 'thumbnail', description: 'Image thumbnail, must be jpeg' do
+    header 'Content-Type', 'image/jpeg'
+    payload Attributor::Tempfile
+    filename values: ['thumb.jpg']
+  end
+
+end
+{% endhighlight %}
+
+
+### Use
```

Comment:
better description:

Using the MulripartArray instances in your controller code: